contracts/src/StabilityPool.sol

```diff
@@ -263,10 +270,16 @@ contract StabilityPool is LiquityBase, IStabilityPool, IStabilityPoolEvents {
 
         _updateDepositAndSnapshots(msg.sender, newDeposit, newStashedColl);
         boldToken.sendToPool(msg.sender, address(this), _topUp);
-        _updateTotalBoldDeposits(_topUp + keptYieldGain, 0);
+        uint256 totalBoldDepositsCached = _updateTotalBoldDeposits(_topUp + keptYieldGain, 0);
         _decreaseYieldGainsOwed(currentYieldGain);
         _sendBoldtoDepositor(msg.sender, yieldGainToSend);
         _sendCollGainToDepositor(collToSend);
+
+        // If there were pending yields and with the new deposit we are reaching the threshold, let’s move the yield to owed
+        uint256 yieldGainsPendingCached = yieldGainsPending;
+        if (yieldGainsPendingCached > 0 && totalBoldDepositsCached >= DECIMAL_PRECISION) {
+            _updateYieldRewardsSum(yieldGainsPendingCached, totalBoldDepositsCached);
+        }
```

Comment:
We must do the same in `withdrawFromSP()`. Reason: a depositor could have enough pending yield gains to put total deposits over the threshold of `DECIMAL_PRECISION`. Remember that yield gains are deposited if they're not claimed.

Reply:
Ah, right!

Reply:
Ah, I see you already fixed it, thanks!